File: hospital-ai/frontend/src/types/emergency.ts
export interface VitalObservation {
  name: string;
  value: string;
  unit: string;
  status: string;
  severity: string;
  message: string;
}
export interface VitalMonitoringResult {
  observed_at: string;
  observations: VitalObservation[];
  abnormal_count: number;
  critical_count: number;
  monitoring_status: string;
  data_source: string;
  limitations: string[];
}
export interface CriticalEvent {
  event_type: string;
  severity: string;
  detected: boolean;
  evidence: string[];
  message: string;
}
export interface CriticalEventDetectionResult {
  events: CriticalEvent[];
  detected_event_count: number;
  critical_event_count: number;
  detection_mode: string;
}
export interface TriageClassification {
  category: string;
  score: number;
  escalation_required: boolean;
  reasons: string[];
  evidence: string[];
  disclaimer: string;
}
export interface ICURequirementResult {
  signal: string;
  score: number;
  confidence: number;
  reasons: string[];
  evidence: string[];
  limitation: string;
}
export interface EmergencyAlert {
  code: string;
  severity: string;
  title: string;
  message: string;
  recommended_next_step: string;
  requires_acknowledgement: boolean;
}
export interface EmergencyAlertGenerationResult {
  alerts: EmergencyAlert[];
  alert_count: number;
  highest_severity: string;
  delivery_mode: string;
}
export interface PatientPriorityRanking {
  priority_score: number;
  priority_level: string;
  rank_basis: string;
  factors: string[];
  disclaimer: string;
}
export interface EmergencyResult {
  id: string;
  patient_id: string;
  processing_job_id?: string | null;
  vital_monitoring_json: VitalMonitoringResult;
  triage_classification_json: TriageClassification;
  critical_event_detection_json: CriticalEventDetectionResult;
  icu_requirement_json: ICURequirementResult;
  emergency_alerts_json: EmergencyAlertGenerationResult;
  patient_priority_json: PatientPriorityRanking;
  summary?: string | null;
  engine: string;
  status: string;
  warnings_json: string[];
  processing_time_ms?: number | null;
  created_at: string;
  updated_at?: string | null;
}
export interface EmergencyStartResult {
  patient_id: string;
  status: string;
  processing_time_ms: number;
  summary: string;
  engine: string;
  warnings: string[];
  vital_monitoring: VitalMonitoringResult;
  triage_classification: TriageClassification;
  critical_event_detection: CriticalEventDetectionResult;
  icu_requirement: ICURequirementResult;
  emergency_alerts: EmergencyAlertGenerationResult;
  patient_priority: PatientPriorityRanking;
  emergency_result: EmergencyResult;
}

File: hospital-ai/backend/app/schemas/emergency.py
"""Pydantic schemas for the Emergency Agent API."""

from __future__ import annotations

from datetime import datetime
from typing import Any, Dict, List
from uuid import UUID

from pydantic import BaseModel, Field

from app.ai.emergency.models import (
    CriticalEventDetectionResult,
    EmergencyAlertGenerationResult,
    ICURequirementResult,
    PatientPriorityRanking,
    TriageClassification,
    VitalMonitoringResult,
)


class EmergencyStartRequest(BaseModel):
    patient_id: UUID


class EmergencyResultOut(BaseModel):
    id: UUID
    patient_id: UUID
    processing_job_id: UUID | None = None
    vital_monitoring_json: Dict[str, Any] = Field(default_factory=dict)
    triage_classification_json: Dict[str, Any] = Field(default_factory=dict)
    critical_event_detection_json: Dict[str, Any] = Field(default_factory=dict)
    icu_requirement_json: Dict[str, Any] = Field(default_factory=dict)
    emergency_alerts_json: Dict[str, Any] = Field(default_factory=dict)
    patient_priority_json: Dict[str, Any] = Field(default_factory=dict)
    summary: str | None = None
    engine: str = "deterministic_emergency_rules_v1"
    status: str = "Completed"
    warnings_json: List[str] = Field(default_factory=list)
    processing_time_ms: int | None = None
    created_at: datetime
    updated_at: datetime | None = None


class EmergencyStartResponse(BaseModel):
    patient_id: UUID
    status: str = "Completed"
    processing_time_ms: int
    summary: str
    engine: str
    warnings: List[str] = Field(default_factory=list)
    vital_monitoring: VitalMonitoringResult
    triage_classification: TriageClassification
    critical_event_detection: CriticalEventDetectionResult
    icu_requirement: ICURequirementResult
    emergency_alerts: EmergencyAlertGenerationResult
    patient_priority: PatientPriorityRanking
    emergency_result: EmergencyResultOut


class EmergencyHistoryItemOut(BaseModel):
    id: UUID
    created_at: datetime
    triage_category: str | None = None
    triage_score: float | None = None
    priority_level: str | None = None
    priority_score: float | None = None
    alert_count: int = 0
    status: str = "Completed"
